backend/internal/controllers/user.go

```diff
@@ -0,0 +1,59 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/TritonSE/words-alive/internal/database"
+	"github.com/TritonSE/words-alive/internal/models"
+)
+
+type UserController struct {
+	// Hold a reference to the database so that all of the functions below can use it
+	// Keeps an authenticator to authenticate token
+	Users database.UserDatabase
+}
+
+func (c *UserController) CreateUser(rw http.ResponseWriter, req *http.Request) {
+	var user models.User
+
+	if err := json.NewDecoder(req.Body).Decode(&user); err != nil {
+		writeResponse(rw, http.StatusBadRequest, "bad input!")
+		return
+	}
+
+	uid, ok := req.Context().Value("user").(string)
+	if !ok {
+		writeResponse(rw, http.StatusInternalServerError, "error")
+		fmt.Println("unable to get user from request context")
+		return
+	}
+	if uid != user.ID {
+		writeResponse(rw, http.StatusForbidden, "Token does not match request body")
+		return
+	}
+
+	_, err := c.Users.FetchUserByEmail(req.Context(), user.Email)
+	if err != nil {
+		// we have created a user with this email before
+		writeResponse(rw, http.StatusBadRequest, "user with email already exists")
+		return
+	}
```

Comment:
I don't think this is correct. Right now this will say "user with email already exists" if we can't run the query at all (database is down, SQL syntax error, etc.).

We should keep this `if err != nil` block and throw a 500 in it, but what's more important is checking if the returned user (the first parameter that you've ignored with a `_`) is non-nil, i.e. a user already exists with that email.